spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigUtils.java

```diff
@@ -271,9 +271,11 @@ private static Map<String, String> decodeData(Map<String, String> data) {
 
 	public static <T> void registerSingle(ConfigurableBootstrapContext bootstrapContext, Class<T> cls, T instance,
 			String name) {
-		bootstrapContext.registerIfAbsent(cls, BootstrapRegistry.InstanceSupplier.of(instance));
-		bootstrapContext.addCloseListener(event -> event.getApplicationContext().getBeanFactory()
-				.registerSingleton(name, event.getBootstrapContext().get(cls)));
+		if (instance != null && !bootstrapContext.isRegistered(cls)) {
```

Comment:
I've done two things here:

- check if `instance` that is passed is null, so that it slightly simplifies callers. 
- check if class is already registered with `!bootstrapContext.isRegistered(cls)`. We already doing that via `regsiterIfAbsent`, but than we were calling `addCloseListener`, no matter if it was registered or not. Not a big deal, but this is cleaner, imo.